orchagent/bufferorch.cpp

```diff
@@ -783,6 +850,9 @@ task_process_status BufferOrch::processEgressBufferProfileList(Consumer &consume
         SWSS_LOG_ERROR("Failed resolving egress buffer profile reference specified for:%s", key.c_str());
         return task_process_status::task_failed;
     }
+
+    setObjectReference(m_buffer_type_maps, consumer.getTableName(), key, buffer_profile_list_field_name, profile_name_list);
```

Comment:
move this just before exiting this function to ensure that the profile is actually applied to the port

Reply:
This is the way in which I implemented it originally.
Why did I change it to the current solution eventually? Because:
- the motivation of the PR is to prevent orchagent from releasing an object with non-zero reference.
- even if the SAI API failed, the dependencies among the orchagent are still there. in this sense, not to update the reference in case of SAI API failure will cause the dependency record be broken, which, I suspect, can cause a subsequent issue
- in case of SAI API failed, no steps can be taken in orchagent to recover it. It will depend on the application to recover it.

In this sense, to do it in the current way is better. How do you think? @neethajohn

Reply:
I am in agreement. @prsunny , @lguohan , do you see any issues with this approach?

Reply:
Since the two returns below doesn't result in retry, lgtm. But just curious, what if there is a retry and `setObjectReference `is called multiple times?

Reply:
Hi
It doesn’t matter how many times the setObjectReference is called because
- ‘set’ is used to record the reference
- operations like insert and remove are idempotent, which means calling it for many times will result in the same result as if calling it for only once.

You can also refer the ***Reference count vs. set*** section in the PR’s description

Meanwhile, if the reference count approach is chosen to record the reference count, a flag is required.